### PR TITLE
Basic login

### PR DESCRIPTION
**Pull request by emilnormann**
Fixed comments.
Helpers not in separate file because of dependencies.
EAS builds locally so should work.

### CODE REVIEW

File: app/(login)/CreateAccountScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  hidePassword: boolean,
+  showWrongPasswords: boolean,
+  fullName: string,
+  email: string,
+  error: string,
+  submitActive: boolean,
+  creatingUser: boolean,
+}
+
+export default class CreateAccountScreen extends Component<Props, State> {
```
Comment (UnluckyBird) [inline app/(login)/CreateAccountScreen.tsx:27]: React does not recommend defining components as classes, but instead as functions per: https://react.dev/reference/react/Component
They have a guide for migration here: https://react.dev/reference/react/Component#alternatives

File: app/(login)/LoginScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+type State = {
+  phoneNumber: string
+  password: string,
+  wrongCredentials: boolean,  
+  hidePassword: boolean,
+  authing: boolean
+}
+
+export default class loginScreen extends Component<Props, State> {
```
Comment (UnluckyBird) [inline app/(login)/LoginScreen.tsx:22]: React does not recommend defining components as classes, but instead as functions per: https://react.dev/reference/react/Component
They have a guide for migration here: https://react.dev/reference/react/Component#alternatives

File: .github/workflows/build.yml
Comment (andreasalstrup) [inline .github/workflows/build.yml:1]: Do not change

File: .github/workflows/update.yml
Comment (andreasalstrup) [inline .github/workflows/update.yml:1]: Do not change

Comment (UnluckyBird) [inline .github/workflows/update.yml:1]: Don't change

File: app/(login)/LoginScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { Text, View, } from '../../components/Themed';
+import { TextInput } from 'react-native-gesture-handler';
+import { ImageBackground, Pressable } from 'react-native';
+import { Component } from 'react';
+import { Link, router } from 'expo-router';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+import { LogoAndName } from '../../components/LogoAndName';
+
+type Props = {
+  text: any
```
Comment (UnluckyBird) [inline app/(login)/LoginScreen.tsx:11]: string

**Review by andreasalstrup — CHANGES_REQUESTED** [review]
LGTM just fix comments

File: constants/Database.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import { IGunInstance, IGunInstanceRoot, IGunUserInstance, ISEA } from 'gun/types';
+const asyncsStore = require('gun/lib/ras.js')
+
+AsyncStorage.clear()
+
+
+//rad asyncstorage adapter, on Android asyncstorage has 6mb limit by default
+const asyncStore = asyncsStore({AsyncStorage});
+let gun = Gun({
+peers: ['https://130.225.39.205:8080/gun'],
```
Comment (andreasalstrup) [inline constants/Database.ts:19]:
```suggestion
peers: ['http://130.225.39.205:8080/gun'],
```

File: app/(login)/CreateAccountScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+          console.log("User somehow doesn't exist");
+      }
+    }
+    else{
+        console.log("Newly created user cannot be authenticated, something went wrong with the database"); 
+        //  Not sure how this should be handled
+    }
+  }
+
+  function validatePhone (phone : string) {
```
Comment (UnluckyBird) [inline app/(login)/CreateAccountScreen.tsx:62]: can be made shorter with `const validatePhone : boolean (phone : string) => phone.length == 8`

File: app/(login)/CreateAccountScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        console.log("Newly created user cannot be authenticated, something went wrong with the database"); 
+        //  Not sure how this should be handled
+    }
+  }
+
+  function validatePhone (phone : string) { 
+    return phone.length == 8 // Danish phonenumbers only atm        
+  }
+
+  function validatePass (pass : string) {
```
Comment (UnluckyBird) [inline app/(login)/CreateAccountScreen.tsx:66]: `const validatePass : boolean (pass: string) => pass.length > 7`

File: app/(login)/CreateAccountScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+  function passwordsMatch (pass : string, repPass : string) {
+    let equal : boolean = pass == repPass
+    let value : boolean = (pass != '' && repPass != '') && (!equal)
+    setShowWrongPasswords(value)
+    return equal
+  }
+
+  function toggleHidePassword () {
```
Comment (UnluckyBird) [inline app/(login)/CreateAccountScreen.tsx:77]: `const toggleHidePassword = () => setHidePassword(!hidePassword)`

**Review by UnluckyBird — APPROVED** [review]

**Pull request merged**